internal/packer/packer.go

```diff
@@ -1467,7 +1467,7 @@ func (pack *Pack) logic(programName string) error {
 		done := measure.Interactively("probing https")
 		remoteScheme, err := httpclient.GetRemoteScheme(updateBaseUrl)
 		done("")
-		if remoteScheme == "https" && !tlsflag.Insecure() {
```

Comment:
I don’t think we can make this change. The -insecure flag is supposed to prevent HTTPS probing to allow folks to first enable TLS.

Based on the error message you provided, it seems to me like the updater.NewTarget() call in line 1494 is what fails, so I’m not sure how your suggested change helps with that…?

Reply:
`GetRemoteScheme` does not return `https` if TLS isn't enabled.

And this insecure check prevents to keep setting the address scheme to `https` when the server actually supports https. So it fallsback to http (when passing insecure), which doesn't work with TLS enabled.

The original https error is unrelated to this fix.